explorer-ui/src/components/Activities/Activities.jsx

```diff
@@ -101,14 +102,11 @@ export default function Activities({currCountry, currProvince, currCity, latitud
 const getXID = async(events) => {
   console.log("events are: ",events)
   try{
-    for (let i = 0; i < events.length; i++) {
-      console.log(i)
-      const xid = await apiGet("xid/" + events[i].xid)
-      setXIDList(xidList => [...xidList, xid]);
-      console.log("xid list so far: ", xidList)
-      console.log("xid is: ", xid); 
-    }
-    return xidList
+      const xidData = await apiGet("xid/" + events.xid)
+      setXID(xidData);
```

Comment:
overwrites the previous array values ?

Reply:
ahhh you're right! not sure how I missed that

Reply:
fixed